### PR TITLE
Update libusb host driver in RTOS device control

### PR DESCRIPTION
**Pull request by lucianomartin**
Part of https://xmosjira.atlassian.net/browse/VFH-111

### CODE REVIEW

File: modules/sw_services/device_control/host/libusb/Win32/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+2. **Download libusb-1.0 Source Code**:
+   - Navigate to the [libusb GitHub repository](https://github.com/libusb/libusb) and download the source code for version 1.0.27 as a ZIP file from [this link](https://github.com/libusb/libusb/archive/refs/tags/v1.0.27.zip).
+
+3. **Extract the ZIP Archive**:
+   - Unzip the downloaded archive to a convenient location on your machine.
+
+4. **Copy Header File**:
+   - Locate the `libusb.h` file in the extracted folder at `libusb-1.0.27\libusb`.
+   - Copy `libusb.h` to the current folder where you want the generated files to reside.
```
Comment (mbanth) [inline modules/sw_services/device_control/host/libusb/Win32/README.md:22]: I suggest removing the word `current`.

File: modules/sw_services/device_control/host/libusb/Win32/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   - Navigate to the `msvc` directory within the extracted `libusb-1.0.27` folder.
+   - Execute the following command to build libusb:
+     ```
+     msbuild -p:PlatformToolset=v143,Platform=win32,Configuration=Release libusb.sln
+     ```
+   - This command compiles the libusb solution using the Visual Studio 2022 (v143) toolset for the Win32 platform in Release configuration.
+
+7. **Copy Generated Library**:
+   - After the build completes, locate the generated `libusb-1.0.lib` file in `libusb-1.0.27\build\v143\Win32\Release\lib`.
+   - Copy `libusb-1.0.lib` to the current folder.
```
Comment (mbanth) [inline modules/sw_services/device_control/host/libusb/Win32/README.md:37]: I suggest changing `current folder` to `folder where you want the generated files to reside`.

File: modules/sw_services/device_control/host/libusb/Win32/libusb-1.0.lib
Comment (mbanth) [inline modules/sw_services/device_control/host/libusb/Win32/libusb-1.0.lib:1]: Do we want to store the `.lib` file in GitHub?  Is it a generated file?

**Review by uvvpavel — APPROVED** [review]
Nice! Looks like we can call the same api on all the platforms. Out of curiosity, is this a 32 or 64-bit binary?

**Review by mbanth — CHANGES_REQUESTED** [review]

**Comment by lucianomartin** [comment]
> Nice! Looks like we can call the same api on all the platforms. Out of curiosity, is this a 32 or 64-bit binary?

It is is 32 bits, but I tested the 64-bit version and it worked fine as well. You just need to re-build the .lib file with the command:

`msbuild -p:PlatformToolset=v143,Platform=x64,Configuration=Release libusb.sln
`

> Nice! Looks like we can call the same api on all the platforms. Out of curiosity, is this a 32 or 64-bit binary?

File: modules/sw_services/device_control/host/libusb/Win32/README.md (hunk 2 above)
Comment (lucianomartin) [reply modules/sw_services/device_control/host/libusb/Win32/README.md]: Updated.

File: modules/sw_services/device_control/host/libusb/Win32/README.md (hunk 1 above)
Comment (lucianomartin) [reply modules/sw_services/device_control/host/libusb/Win32/README.md]: Updated.

File: modules/sw_services/device_control/host/libusb/Win32/libusb-1.0.lib
Comment (lucianomartin) [reply modules/sw_services/device_control/host/libusb/Win32/libusb-1.0.lib:1]: This is what we usually do, the alternative is for the user to recompile the file, but it is not a simple task.
All the .lib files we store in GitHub can be regenerated, but sharing a compiled version is the most convenient solution for the users. 
Unless we want to store them somewhere else.

**Review by mbanth — APPROVED** [review]

**Pull request merged**